haystack/components/converters/pypdf.py

```diff
@@ -40,7 +40,8 @@ class DefaultConverter:
     The default converter class that extracts text from a PdfReader object's pages and returns a Document.
     """
 
-    def convert(self, reader: "PdfReader") -> Document:
+    @staticmethod
+    def convert(reader: "PdfReader") -> Document:
```

Comment:
This is a breaking change, is this necessary?

Reply:
not necessary, I was reviewing it, and noticed this method should be static, why is it breaking? I've missed it

Reply:
I agree it should be static but we should have thought of this when adding this Component. We should have made it `_convert` from the start.

As of now it's part of the public interface of the class and converting it simply to `staticmethod` can break existing uses so if we change it we must follow the usual deprecation policy.

Reply:
also don't see why this would be breaking

Reply:
Eeeeh, true. You can still call static methods from the instance, though not the most elegant thing.

Reply:
ok, I will revert it, since @wochinge would like this in the next release, and to deprecate it, we would first to need to make a release with the warning and then the actual deprecation

Reply:
Okay, I don't mind but I don't see why doing a non breaking change would cause so much aftermath. Nothing changes for the user at all 🤷🏻

Reply:
It's best if we behave conservatively for this kind of changes in any case.

It might seem like it doesn't break anything but I very much prefer we don't risk it and follow the deprecation policy in any case, and even make the method private.

It's always annoying having to backtrack a change after a release and I don't want anyone dealing with that.